server/controllers/wishlistController.js

```diff
@@ -0,0 +1,202 @@
+
+  try {
+    Wishlist.find({ family: familyId }).exec((err, wishlist) => {
+      if (err) {
+        res.status(500).json({ Error: err });
+      }
+
+      wishlist[0].list.push(item);
+
+      wishlist[0].totalListCost = _getWishlistCost(wishlist[0].list);
+
+      wishlist[0].save((err, wishlist) => {
+        if (err) {
+          res.status(500).json({ Error: err });
+        }
+
+        res.status(201).json({
+          message: 'Successfully added item to wishlist',
+          wishlist: wishlist,
+        });
+      });
+    });
+  } catch (err) {
+    res.status(500).json({ Message: 'TEMP ERROR TEST' });
+  }
+};
+
+exports.create = (req, res) => {
+  // Bad request if no body is sent
+  if (Object.keys(req.body).length === 0) {
+    return res.status(400).json({
+      Error: 'Body missing data',
+    });
+  }
+
+  const familyId = req.params.familyId;
+  const newWishlist = req.body.list;
+
+  // Check newWishlist object type is an Array
+  // Single items should still reside in an Array
+
+  try {
+    Wishlist.where({ family: familyId }).exec((err, wishlist) => {
+      if (err) {
+        res.status(500).json({ Error: err });
+      }
+
+      if (wishlist[0].list !== []) {
+        return res.status(200).json({
+          Message: 'Unable to create wishlist. Wishlist already exists.',
+        });
+      }
+
+      wishlist[0].list = newWishlist;
+
+      wishlist[0].totalListCost = _getWishlistCost(wishlist[0].list);
+
+      wishlist[0].save((err, wishlist) => {
+        if (err) {
+          res.status(500).json({ Error: err });
+        }
+
+        res.status(200).json({
+          message: 'Successfully added new wishlist to family.',
+          wishlist: wishlist,
+        });
+      });
+    });
+  } catch (err) {
+    res.status(500).json({ Message: 'TEMP ERROR TEST' });
+  }
+};
+
+exports.delete = (req, res) => {
+  res.send('NOT YET IMPLEMENTED');
+};
+
+exports.removeItem = (req, res) => {
+  // Bad request if no body is sent
+  if (Object.keys(req.body).length === 0) {
+    return res.status(400).json({
+      Error: 'Body missing data',
+    });
+  }
+
+  const familyId = req.params.familyId;
+  const { itemName } = req.body.item;
+  const itemId = req.body.itemId;
+
+  try {
+    Wishlist.where({ family: familyId })
+      .findOne({
+        'list.itemName': itemName,
+        'list._id': itemId,
+      })
+      .exec((err, wishlist) => {
+        if (err) {
+          res.status(500).json({ Error: err });
+        }
+
+        if (!wishlist) {
+          return res
+            .status(400)
+            .send('No wishlist found containing that itemId');
+        }
+
+        wishlist.list = wishlist.list.filter(
+          item => item._id.toString() !== itemId
+        );
+
+        wishlist.totalListCost = _getWishlistCost(wishlist.list);
+
+        wishlist.save((err, wishlist) => {
+          if (err) {
+            res.status(500).json({ Error: err });
+          }
+
+          res.status(200).json({
```

Comment:
from MDN: 
> The successful result of a PUT or a DELETE is often not a 200 OK but a 204 No Content (or a 201 Created when the resource is uploaded for the first time).

Also, don't forget to put he body of newly created/modified list in the response

Reply:
Oh okay, return the body of what was added? 

Currently, these routes are just returning the entire new family wishlist. I can change it.

Reply:
@jwu910 I believe returning the entire new thing is the norm. so even if you had a wishlist with 10 items and you added item 11, you would return all 11 items and not just the 1 change

Reply:
@rainrivas Oh we're good then. I'll just make sure the status codes are updated.